src/Psalm/Issue/InvalidIntersectionType.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Psalm\Issue;
+
+final class InvalidIntersectionType extends CodeIssue
+{
+    public const ERROR_LEVEL = 6;
+    public const SHORTCODE = 283;
```

Comment:
@orklah do you know why we have a gap in shortcodes? Usually the think to do is run `bin/max_used_shortcode.php` then add 1 to it when adding a new issue.

Reply:
I've bumped the shortcode using the command 👍

The `ERROR_LEVEL` was a copy-paste. Happy to change it, I'm not that familiar with the scale!

Reply:
> I'm not that familiar with the scale!

Neither am I! :laughing:

I mostly run Psalm at level 1 on newer projects, so I don't feel like I have a good idea on what should go where. There's some documentation [here](https://psalm.dev/docs/running_psalm/error_levels/) (that I've been meaning to write a test for, I'm sure a bunch of issues are missing from those lists).

Reply:
Basically, the scale is 8 = super bad, will directly lead to fatal errors or will prevent Psalm from analyzing at a deep level (for example a missing class), 1= Mixed issues (can't say if bad or not)

In between you have the gradient. For example, `InvalidArgument` will be at level 6, `PossiblyInvalidArgument` will be at level 3.

For me, this particular issue is almost the same as `NoValue` (basically, every possible Type has been excluded so all that's left is `Never`: there is no type that will fit that particular variable/expression). Except that NoValue is emitted when using a `Never` expression, while this would be emitted while encountering one. However, NoValue is at `-1` (usually used when a particular issue is not dependent on level, for example when it's behind a config flag so it's always reported when triggered). NoValue used to be reported rarely (namely when using the return of a never-returning function) but it took a much bigger role recently.

Maybe we should decide a level and stick NoValue and this on the same level?